cpp/src/io/fst/agent_dfa.cuh

```diff
@@ -448,11 +448,12 @@ struct AgentDFA {
             typename SymbolMatcherT,
             typename CallbackOpT,
             int32_t IS_FULL_BLOCK>
-  __device__ __forceinline__ static void ThreadParse(SymbolMatcherT const& symbol_matcher,
-                                                     CharT const* chars,
-                                                     SymbolIndexT const& max_num_chars,
-                                                     CallbackOpT callback_op,
-                                                     cub::Int2Type<IS_FULL_BLOCK> /*IS_FULL_BLOCK*/)
+  __device__ __forceinline__ static void ThreadParse(
+    SymbolMatcherT const& symbol_matcher,
+    CharT const* chars,
+    SymbolIndexT const& max_num_chars,
+    CallbackOpT callback_op,
+    cuda::std::integral_constant<int, IS_FULL_BLOCK> /*IS_FULL_BLOCK*/)
```

Comment:
```suggestion
    cuda::std::bool_constant<IS_FULL_BLOCK> /*IS_FULL_BLOCK*/)
```
Is `bool_constant` available in CCCL?

Reply:
Oh yeah it is, and I always forget it because for a long time it was behind C++14

Will adopt

Reply:
The change looks great to me. I fully support this adoption as it applies to detailed functions rather than public ones. In libcudf, only changes to public APIs are considered breaking changes.

Reply:
@PointKernel is there something wrong in CI because [those](https://github.com/rapidsai/cudf/actions/runs/13376114806/job/37355755788#step:9:3807) errors seem unrelated:
>$SRC_DIR/cpp/src/io/utilities/data_sink.cpp: In constructor 'cudf::io::file_sink::file_sink(const string&)':
$SRC_DIR/cpp/src/io/utilities/data_sink.cpp:40:32: error: 'class kvikio::FileHandle' has no member named 'is_compat_mode_preferred'
   40 |                   _kvikio_file.is_compat_mode_preferred() ? "on" : "off");

Reply:
This should be related to the breaking change from https://github.com/rapidsai/kvikio/pull/608